plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts

```diff
@@ -26,18 +28,39 @@ import {
 } from './types';
 import { runDockerContainer, runCommand } from './helpers';
 
-const commandExists = require('command-exists-promise');
+type TechdocsGeneratorOptions = {
+  // This option enables users to configure if they want to use TechDocs container
+  // or generate without the container.
+  // This is used to avoid running into Docker in Docker environment.
+  runGeneratorIn: string;
+};
+
+const createStream = (): [string[], PassThrough] => {
+  const log = [] as Array<string>;
+
+  const stream = new PassThrough();
+  stream.on('data', chunk => {
+    const textValue = chunk.toString().trim();
+    if (textValue?.length > 1) log.push(textValue);
+  });
+
+  return [log, stream];
+};
```

Comment:
Looks like a useful utility function, that can live out of here.

Reply:
It might be useful in another context, but i think i'd prefer it to be here for now. I think we should look at providing a stream in the router for the build job instead. That way we could allow any error to be handled in a more generic and flexible way.